src/core/sagas/abuse.js

```diff
@@ -42,6 +48,27 @@ export function* reportAddon({
   }
 }
 
+export function* reportAddonViaFirefox({
+  payload: { addon },
+}: InitiateAddonAbuseReportViaFirefoxAction): Saga {
+  const abuseReported = yield reportAbuse(addon.guid);
```

Comment:
What happens if this `throws`?

Reply:
We kind of discussed this a bit in IRC, but I see now it's more complicated than I have dealt with. We discussed that we can assume that the underlying `mozAddonManager.reportAbuse` won't throw/reject, but you're correct that our own `reportAbuse` can. It's designed to reject if the API isn't available, but we are also checking for that availability before ever firing off this saga, so, as long as it's used as intended, our own `reportAbuse` should never reject either, but there's no guarantee of that.

I need to think a bit more about what we would want to do if is does reject.

Reply:
What I decided to do was simply log a warning if the promise rejects, and then dispatch `finishAddonAbuseReportViaFirefox` to reset the UI. This would mean that if the saga was somehow called despite the API not being available, a user would see the button disabled and then re-enabled, but a message would be logged to the console explaining that something went wrong. Hopefully that would be enough to allow us to fix the situation it someone makes a coding error. I think doing anything more than that might be trying to make the saga "too smart".